packages/share_plus/share_plus/lib/share_plus.dart

```diff
@@ -79,4 +81,57 @@ class Share {
       sharePositionOrigin: sharePositionOrigin,
     );
   }
+
+  /// Behaves exactly like [share] while providing feedback on how the user
+  /// interacted with the share-sheet. Until the returned future is completed,
+  /// any other call to any share method that returns a result might result in
+  /// a [PlatformException].
+  ///
+  /// Because IOS and Android provide different feedback on share-sheet interaction,
+  /// a result on IOS will be more specific than on Android. While IOS can detect if
+  /// the user actually completed his selected action or aborted it midway, Android
+  /// only records if the user selected an action or outright dismissed the share-sheet.
+  ///
+  /// **Currently only implemented on IOS & Android.**
```

Comment:
What would happen if a dev calls this method from desktop or web?

Reply:
It would raise an UnimplementedError